tests/test_pf7.py

```diff
@@ -0,0 +1,433 @@
+        test_pf7_extended = Pf7(self.test_data_path, data_config=self.test_config_path)
+        test_pf7_extended.extended_variant_fields = {"AN": [DIM_VARIANT]}
+        test_pf7_extended.extended_calldata_variables = {
+            "GQ": [DIM_VARIANT, DIM_SAMPLE]
+        }
+        actual_extended = test_pf7_extended._add_extended_data(
+            self.test_zarr_root, True, "native", {}
+        )
+        self.assertEqual(
+            list(actual_extended.keys()),
+            [
+                "call_GQ",
+                "variant_AN",
+            ],
+        )
+
+        values = [actual_extended[k] for k in actual_extended]
+        for value in values:
+            assert isinstance(value[1], da.Array)
+        dimensions = [i[0] for i in values]
+        self.assertEqual(
+            dimensions,
+            [
+                ["variants", "samples"],
+                ["variants"],
+            ],
+        )
+
+    @patch("malariagen_data.pf7.Pf7.open_zarr")
+    def test_variant_calls_default(self, mock_open_zarr):
+        mock_open_zarr.return_value = self.test_zarr_root
+        ds = self.test_pf7_class.variant_calls(inline_array=True, chunks="native")
+        mock_open_zarr.assert_called_once_with()
+        coords = list(ds.coords.keys())
+        variables = list(ds.keys())
+        self.assertEqual(coords, ["variant_position", "variant_chrom", "sample_id"])
+        self.assertEqual(
+            variables,
+            [
+                "variant_allele",
+                "variant_filter_pass",
+                "variant_is_snp",
+                "variant_numalt",
+                "variant_CDS",
+                "call_genotype",
+                "call_AD",
+            ],
+        )
+
+    @patch("malariagen_data.pf7.Pf7.open_zarr")
+    def test_variant_calls_extended(self, mock_open_zarr):
+        mock_open_zarr.return_value = self.test_zarr_root
+        test_pf7_class_extended = Pf7(
+            self.test_data_path,
+            data_config=self.test_config_path,
+        )
+        test_pf7_class_extended.extended_variant_fields = {"AN": [DIM_VARIANT]}
+        test_pf7_class_extended.extended_calldata_variables = {
+            "GQ": [DIM_VARIANT, DIM_SAMPLE]
+        }
+        ds = test_pf7_class_extended.variant_calls(
+            extended=True,
+            inline_array=True,
+            chunks="native",
+        )
+        mock_open_zarr.assert_called_once_with()
+        coords = list(ds.coords.keys())
+        variables = list(ds.keys())
+        self.assertEqual(coords, ["variant_position", "variant_chrom", "sample_id"])
+        self.assertEqual(
+            variables,
+            [
+                "variant_allele",
+                "variant_filter_pass",
+                "variant_is_snp",
+                "variant_numalt",
+                "variant_CDS",
+                "call_genotype",
+                "call_AD",
+                "call_GQ",
+                "variant_AN",
+            ],
+        )
```

Comment:
This tests the `variant_calls()` public method with `extended=True` :+1: 

Although again you could do this without any mocking, similar to how `test_sample_metadata_returns_extpected_df` works without any mocking.